carma_wm/src/WorldModelUtils.cpp

```diff
@@ -0,0 +1,126 @@
+{
+  auto possible_lanelets = getLaneletsFromPoint(semantic_map, point, n);
+  for (auto llt : possible_lanelets)
+  {
+    semantic_map->laneletLayer.get(llt.id());
+  }
+  return possible_lanelets;
+}
+
+std::vector<lanelet::Lanelet> nonConnectedAdjacentLeft(const lanelet::LaneletMapPtr& semantic_map, const lanelet::BasicPoint2d& point,
+                                                                    const unsigned int n)
+{
+  auto possible_lanelets = nonConnectedAdjacentLeft(semantic_map, point, n);
+  for (auto llt : possible_lanelets)
+  {
+    semantic_map->laneletLayer.get(llt.id());
+  }
+  return possible_lanelets;
+}
+
+std::vector<lanelet::ConstLanelet> nonConnectedAdjacentLeft(const lanelet::LaneletMapConstPtr& semantic_map, const lanelet::BasicPoint2d& input_point,
+                                                                    const unsigned int n)
+{
+  // Check if the map is loaded yet
+  if (!semantic_map || semantic_map->laneletLayer.size() == 0)
+  {
+    throw std::invalid_argument("Map is not set or does not contain lanelets");
+  }
+  
+  auto input_lanelets = getLaneletsFromPoint(semantic_map, input_point);
+  if (input_lanelets.empty())
+  {
+    throw std::invalid_argument("Input point x: " + std::to_string(input_point.x()) + ", y: " + std::to_string(input_point.y()) + " is not in the map");
+  }
+  auto input_lanelet = input_lanelets[0]; 
+
+  ROS_DEBUG_STREAM("right ls size: " << input_lanelet.leftBound2d().size());
+  
+  auto point_downtrack = carma_wm::geometry::trackPos(input_lanelet, input_point).downtrack;
+  ROS_DEBUG_STREAM("point_downtrack: " << point_downtrack);
+  
+  auto point_downtrack_ratio = point_downtrack / carma_wm::geometry::trackPos(input_lanelet, input_lanelet.centerline().back().basicPoint2d()).downtrack;
+  ROS_DEBUG_STREAM("point_downtrack_ratio: " << point_downtrack_ratio);
+
+  auto point_on_ls = input_lanelet.leftBound2d()[std::round((input_lanelet.leftBound2d().size()- 1) * point_downtrack_ratio) ];
+
+  ROS_DEBUG_STREAM("x: " << point_on_ls.basicPoint2d().x() <<  ", y: " << point_on_ls.basicPoint2d().y());
+
+  // point_on_opposite_lane coord is acquired by extrapolating the line from input_point to point_on_ls with same distance.
+  // threfore point_on_opposite_lane.x() = input_point.x() + 2dx, where dx = point_on_ls.x() - input_point.x(). Here, one of input_point.x() cancels out, results in:
+  auto point_on_opposite_lane = lanelet::BasicPoint2d{2 * point_on_ls.x() - input_point.x(), 2 * point_on_ls.y() - input_point.y()};
+
+  ROS_DEBUG_STREAM("x: " << point_on_opposite_lane.x() <<  ", y: " << point_on_opposite_lane.y());
+
+  auto opposite_lanelets = getLaneletsFromPoint(semantic_map, point_on_opposite_lane, n);
+
+  for (auto llt: opposite_lanelets)
+  {
+      ROS_DEBUG_STREAM("potential_opposite_lanelets llt: "  << llt.id());
+  }
+
+  return opposite_lanelets;
+}
+
+uint32_t get32BitId(uint16_t intersection_id, uint8_t signal_group_id)
+{
+  uint32_t temp;
```

Comment:
This must be initialized to 0 see https://sonarcloud.io/project/issues?id=usdot-fhwa-stol_CARMAPlatform&issues=AXrp-ykyxsPBQOrooHWR&open=AXrp-ykyxsPBQOrooHWR&pullRequest=1358